packages/components/app/styles/components/dropdown.scss

```diff
@@ -408,8 +416,86 @@ $hds-dropdown-toggle-border-radius: 5px;
   }
 }
 
+// HDS::DROPDOWN::LIST-ITEM::TITLE
+
 .hds-dropdown-list-item--title {
   padding: 10px 16px 4px;
   color: var(--token-color-foreground-strong);
 }
 
+// HDS::DROPDOWN::LIST-ITEM::CHECKMARK
+
+.hds-dropdown-list-item--checkmark {
+  a.hds-dropdown-list-item__interactive,
+  button.hds-dropdown-list-item__interactive {
+    // leave room for the checkmark to align the result count on non-selected items
+    padding-right: 25px; // 9px default + 16px for the 'check' FlightIcon
+
+    // disable background on focus and active states
+    &:focus,
+    &:active,
+    &.mock-focus,
+    &.mock-active {
+      color: var(--token-color-foreground-action-active);
+
+      &::after {
+        background-color: inherit;
+      }
+    }
+  }
+
+  .hds-dropdown-list-item__interactive-text {
+    flex: 1;
+  }
+}
+
+.hds-dropdown-list-item--checkmark-selected {
+  a.hds-dropdown-list-item__interactive,
+  button.hds-dropdown-list-item__interactive {
+    padding-right: 9px; // reset to default
+    color: var(--token-color-foreground-action);
+  }
+}
+
+.hds-dropdown-list-item__checkmark {
+  align-self: flex-start;
+  margin-top: 0.125rem;
```

Comment:
if you use a container you don't need this padding anymore (you can set the height directly to the desired value, which is `20px`, the line-height of the main text)